backend/internal/repository/starrocks_repository.go

```diff
@@ -298,6 +300,45 @@ func (r *StarrocksRepository) AggregateOccurrences(seqId int, userQuery types.Ag
 	return aggregation, err
 }
 
+func prepareStatQuery(seqId int, userQuery types.StatisticsQuery, r *StarrocksRepository) (*gorm.DB, int, error) {
+	part, err := r.GetPart(seqId)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error during partition fetch %w", err)
+	}
+	tx := addImplicitOccurrencesFilters(seqId, r, part)
+	if userQuery != nil {
+		tx = joinWithVariants(userQuery, tx)
+		if userQuery.HasFieldFromTables(types.ConsequenceFilterTable) || userQuery.HasFieldFromTables(types.GenePanelsTables...) {
+			joinClause := "LEFT JOIN consequences_filter cf ON cf.locus_id=o.locus_id AND cf.part = o.part"
+			tx = tx.Joins(joinClause)
+			selectedPanelsTables := getDistinctTablesFromFields(userQuery.GetFieldsFromTables(types.GenePanelsTables...))
+			for _, panelsTable := range selectedPanelsTables {
+				tx = tx.
+					Joins(fmt.Sprintf("LEFT JOIN %s %s ON %s.symbol=cf.symbol", panelsTable.Name, panelsTable.Alias, panelsTable.Alias))
+			}
+		}
+		addWhere(userQuery, tx)
+
+	}
+	return tx, part, nil
```

Comment:
We are nos supposed to have the same query that aggregate except n groupby and min, max instead of count()?

Reply:
duplicated function removed